pkg/kubectl/cmd/util/factory_builder.go

```diff
@@ -74,25 +92,12 @@ func (f *ring2Factory) PrinterForMapping(cmd *cobra.Command, mapping *meta.RESTM
 		}
 
 		printer = printers.NewVersionedPrinter(printer, mapping.ObjectConvertor, version, mapping.GroupVersionKind.GroupVersion())
+
 	} else {
-		// Some callers do not have "label-columns" so we can't use the GetFlagStringSlice() helper
-		columnLabel, err := cmd.Flags().GetStringSlice("label-columns")
-		if err != nil {
-			columnLabel = []string{}
-		}
-		printer, err = f.clientAccessFactory.Printer(mapping, printers.PrintOptions{
-			NoHeaders:          GetFlagBool(cmd, "no-headers"),
-			WithNamespace:      withNamespace,
-			Wide:               GetWideFlag(cmd),
-			ShowAll:            GetFlagBool(cmd, "show-all"),
-			ShowLabels:         GetFlagBool(cmd, "show-labels"),
-			AbsoluteTimestamps: isWatch(cmd),
-			ColumnLabels:       columnLabel,
-		})
-		if err != nil {
-			return nil, err
+		if humanReadablePrinter, ok := printer.(*printers.HumanReadablePrinter); ok {
+			printersinternal.AddHandlers(humanReadablePrinter)
+			printer = printers.ResourcePrinter(humanReadablePrinter)
```

Comment:
Thanks for adding the comment. It looks good now. As per test-run and removing the label, @mengqiy took care of it.